src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,138 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ELEMENT_NOT_FOUND = -1;
+    private static final double CAPACITY_MULTIPLIER = 1.5;
+
+    private int size;
+    private T[] innerArray;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        this.size = 0;
+        this.innerArray = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
+        if (size + 1 >= innerArray.length) {
+            addCapacity();
+        }
 
+        innerArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexValidity(index, true);
+
+        if (size + 1 >= innerArray.length) {
+            addCapacity();
+        }
```

Comment:
Don't create repeating code. The logic for checking capacity and expanding the innerArray is repeated in both add methods. Consider moving this logic to a separate method to adhere to the DRY principle.